cpp/velox/shuffle/VeloxShuffleWriter.cc

```diff
@@ -1275,9 +1274,17 @@ arrow::Status VeloxShuffleWriter::splitFixedWidthValueBuffer(const velox::RowVec
 
     VS_PRINTLF(bytesPerRow);
 
-    uint64_t preAllocRowCnt = options_.offheap_per_task > 0 && bytesPerRow > 0
-        ? options_.offheap_per_task / bytesPerRow / numPartitions_ >> 2
-        : options_.buffer_size;
+    // remove the cache memory size since it can be spilled.
+    // we can't use pool_->bytes_allocated() here because it's arrow::default_pool(). we should define a
+    // pool for spill buffer, just like the pool for split buffer
+    // memLimit+=pool_->bytes_allocated();
+    memLimit += totalCachedPayloadSize();
```

Comment:
@marin-ma can we use cached buffer memory pool directly?

Reply:
Yes. Removed in #3159
https://github.com/oap-project/gluten/pull/3159/files#diff-55954c774f5d66d0d359c3c7baa29be65f75198b30fd27603b0456ccd11b6a79R1786